spinalcordtoolbox/scripts/sct_compute_snr.py

```diff
@@ -176,26 +203,68 @@ def main(argv=None):
         im_stdnonzero.save(fname_stdnonzero, dtype=np.float32)
         # Compute SNR in ROI
         if fname_mask:
-            mean_in_roi = np.average(data_mean[mask_std_nonzero], weights=mask[mask_std_nonzero])
-            std_in_roi = np.average(data_std[mask_std_nonzero], weights=mask[mask_std_nonzero])
-            snr_roi = mean_in_roi / std_in_roi
-            # snr_roi = np.average(snr_map[mask_std_nonzero], weights=mask[mask_std_nonzero])
+            snr_roi = np.average(snr_map[mask_std_nonzero], weights=mask[mask_std_nonzero])
 
     elif method == 'diff':
+        # Check user selected exactly 2 volumes for this method.
+        if not len(index_vol) == 2:
+            raise ValueError(f"Number of selected volumes: {len(index_vol)}. The method 'diff' should be used with "
+                             f"exactly 2 volumes. You can specify the number of volumes with the flag '-vol'.")
         data_2vol = np.take(data, index_vol, axis=3)
-        # Compute mean in ROI
+        # Compute mean across the two volumes
         data_mean = np.mean(data_2vol, axis=3)
-        mean_in_roi = np.average(data_mean, weights=mask)
+        # Compute mean in ROI for each z-slice, if the slice in the mask is not null
+        mean_in_roi = [np.average(data_mean[..., iz], weights=mask[..., iz])
+                       for iz in range(nz) if np.any(mask[..., iz])]
         data_sub = np.subtract(data_2vol[:, :, :, 1], data_2vol[:, :, :, 0])
-        _, std_in_roi = weighted_avg_and_std(data_sub, mask)
-        # Compute SNR, correcting for Rayleigh noise (see eq. 7 in Dietrich et al.)
-        snr_roi = (2 / np.sqrt(2)) * mean_in_roi / std_in_roi
+        # Compute STD in the ROI for each z-slice. The "np.sqrt(2)" results from the variance of the subtraction of two
```

Comment:
@joshuacwnewton would you prefer I add your explanation? I thought redirecting to this issue would suffice:

> # More context in: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3481

Reply:
> This might not be strictly necessary, but I wanted to confirm my understanding by writing things out. 😅

ah! i'm just seeing this. Indeed, your understanding is 100% correct 😊

Reply:
Basically just wanted to leave some documentation on this PR for future reference, hehe. :smile: 

I've marked these comments as resolved, and so I think this PR is good to go!